### PR TITLE
ci: init jenkins pipeline

### PR DESCRIPTION
**Pull request by ernestojeda**
- Init files for the Jenkins pipeline
- Align to Go 1.16 to be consistent with other repos
- Add go mod tidy to be consistent with other repos

Signed-off-by: Ernesto Ojeda <ernesto.ojeda@intel.com>

### CODE REVIEW

**Review by bill-mahoney — APPROVED** [review]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -44,9 +45,9 @@ ENV APP_PORT=49994
...
 WORKDIR /
-COPY --from=builder /go/src/github.com/edgexfoundry/device-gpio/LICENSE /
-COPY --from=builder /go/src/github.com/edgexfoundry/device-gpio/Attribution.txt /
-COPY --from=builder /go/src/github.com/edgexfoundry/device-gpio/cmd/ /
+COPY --from=builder /device-gpio/LICENSE /
+COPY --from=builder /device-gpio/Attribution.txt /
+COPY --from=builder /device-gpio/cmd/ /
 
 ENTRYPOINT ["/device-gpio"]
 CMD ["-cp=consul://edgex-core-consul:8500", "--registry", "--confdir=/res"]
```
Comment (lenny-intel) [inline Dockerfile:53]:
```suggestion
CMD ["-cp=consul.http://edgex-core-consul:8500", "--registry", "--confdir=/res"]
```

**Review by lenny-intel — CHANGES_REQUESTED** [review]
Why are the two shell scripts empty

**Comment by ernestojeda** [comment]
> Why are the two shell scripts empty

They were not executable so they were failing in the pipeline. I just did a `chmod +x` on them. I think that is what GitHub is showing

File: Dockerfile (hunk 1 above)
Comment (ernestojeda) [reply Dockerfile]: Done

**Review by soda480 — APPROVED** [review]
LGTM

File: Dockerfile [hunk 2]
```diff
@@ -44,9 +45,9 @@ ENV APP_PORT=49994
...
-COPY --from=builder /go/src/github.com/edgexfoundry/device-gpio/LICENSE /
-COPY --from=builder /go/src/github.com/edgexfoundry/device-gpio/Attribution.txt /
-COPY --from=builder /go/src/github.com/edgexfoundry/device-gpio/cmd/ /
+COPY --from=builder /device-gpio/LICENSE /
+COPY --from=builder /device-gpio/Attribution.txt /
+COPY --from=builder /device-gpio/cmd/ /
 
 ENTRYPOINT ["/device-gpio"]
-CMD ["-cp=consul://edgex-core-consul:8500", "--registry", "--confdir=/res"]
+CMD ["-cp=consul.http://edgex-core-consul:8500", "--registry", "--confdir=/res"]
```
Comment (iain-anderson) [inline Dockerfile:53]: Should stay as just "consul"

Comment (ernestojeda) [reply Dockerfile:53]: This was @lenny-intel's suggestion https://github.com/edgexfoundry/device-gpio/pull/3#discussion_r653071433

image

Comment (ernestojeda) [reply Dockerfile:53]: @iain-anderson @lenny-intel  I can go either way. Let me know what this value should be.

Comment (iain-anderson) [reply Dockerfile:53]: Don't mind, @lenny-intel suggested to not use the http in another context.

Comment (lenny-intel) [reply Dockerfile:53]: @iain-anderson , specifying it in the Type field in TOML is non-standard (not supported in Go services) and this is for Configuration Provider, not Registry. More of a consistency thing as all App/core/support and most Device services have it this way.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**